### ISSUE TITLE
Running GenerateResourcesAndImage causes error, hangs if automated - win2022

### ISSUE THREAD

**Issue by jeffpuckett** [actions/virtual-environments#5630]
### Description

A couple of notes. I'm trying to automate the building of an Azure Pipelines Agent image and I'm using GenerateResourcesAndImage to do so. If that is the wrong place to start let me know. I'm running this in an Azure Pipeline using Windows hosted agents. Basically this (with the steps to get the code and install az from psgallery

        - powershell: |
            Set-Location $(Agent.BuildDirectory)\virtual-environments
            Import-Module .\helpers\GenerateResourcesAndImage.ps1
            GenerateResourcesAndImage -SubscriptionId "*****" -ResourceGroupName "$(resourceGroup)" -ImageGenerationRepositoryRoot "$pwd" -ImageType Windows2022 -AzureLocation "East US" -AzureClientId "*****" -AzureClientSecret "*****" -AzureTenantId '*****' -Force
          displayName: Build the Image

1. Currently I'm getting this error

==> vhd: Provisioning with powershell script: D:\a\1\virtual-environments\images\win/scripts/Installers/Install-AzureCli.ps1
    vhd: Install the latest Azure CLI release
    vhd: Downloading azure-cli.msi...
    vhd: Downloading package from: https://aka.ms/installazurecliwindows to path C:\Users\packer\AppData\Local\Temp\azure-cli.msi .
    vhd: Package downloaded successfully in 3.75 seconds
    vhd: Starting Install azure-cli.msi...
    vhd: Non zero exit code returned by the installation process: 1618
    vhd: Total time elapsed: 4.08 seconds
==> vhd: Script exited with non-zero exit status: 1618.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)? 
##[error]The operation was canceled.
Finishing: Build the Image

2. Is there a way to make this a silent success or failure? Or have a flag that defaults to "clean up and exit?" As it stands when this happens the job times out after 6 hours and there is still a temporary resource group left. 

### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

windows-latest

### Repro steps

basically run the GenerateResourcesAndImage.ps1 in an azure pipeline. I would like it if

1. It succeeded, and
2. If there is a failure, it didn't hang

**Comment by al-cheb** [comment]
Hey @jeffpuckett.
You should remove `-on-error=ask` in https://github.com/actions/virtual-environments/blob/12cbf387cabe5494daf18f449097bc6eb899cb27/helpers/GenerateResourcesAndImage.ps1#L284

**Comment by jeffpuckett** [comment]
so should i submit a PR then? I would prefer to pull from the official source and not have to maintain a fork.

**Comment by al-cheb** [comment]
Merged https://github.com/actions/virtual-environments/pull/5635 PR.
